src/coreclr/jit/emitarm64.cpp

```diff
@@ -14662,6 +14804,219 @@ void emitter::emitIns_Call(EmitCallType          callType,
     return PREDICATE_NONE;
 }
 
+/*****************************************************************************
+ *
+ *  Returns true if the SVE instruction has a LSL addr.
+ *  This is for formats that have [<Xn|SP>, <Zm>.T, <mod> #N], [<Xn|SP>, <Xm>, LSL #N]
+ */
+/*static*/ bool emitter::insSveIsLsl(instruction ins, insFormat fmt)
+{
+    switch (fmt)
+    {
+        case IF_SVE_JD_4A:
+            switch (ins)
+            {
+                case INS_sve_st1h:
+                    return true;
+
+                default:
+                    break;
+            }
+            return false;
+
+        case IF_SVE_JD_4B:
+            switch (ins)
+            {
+                case INS_sve_st1w:
+                    return true;
+
+                default:
+                    break;
+            }
+            assert(!"Invalid instruction");
+            return false;
+
+        case IF_SVE_HW_4B:
+            switch (ins)
+            {
+                case INS_sve_ld1h:
+                case INS_sve_ld1sh:
+                case INS_sve_ldff1h:
+                case INS_sve_ldff1sh:
+                case INS_sve_ld1w:
+                case INS_sve_ldff1w:
+                    return true;
+
+                default:
+                    break;
+            }
+            return false;
+
+        default:
+            break;
+    }
+
+    return false;
```

Comment:
remove the `return false` from other places and just have it here?

```c++
            assert(!"Invalid instruction");
            return false;
```

Reply:
this will make sure that `insSveIsLsl` is called only for appropriate formats/instructions.

Reply:
I actually do not want to assert for specific formats in `insSveIsLsl`. The `assert(!"Invalid instruction")` is just for that specific format since it only has one instruction associated with it.

Reply:
I'll just remove the assert altogether and just add a break.